x-pack/plugins/ml/public/application/components/annotations/annotations_table/annotations_table.js

```diff
@@ -447,38 +482,116 @@ export class AnnotationsTable extends Component {
       });
     }
 
-    columns.push({
-      align: RIGHT_ALIGNMENT,
-      width: '60px',
-      name: i18n.translate('xpack.ml.annotationsTable.actionsColumnName', {
-        defaultMessage: 'Actions',
-      }),
-      actions,
-    });
-
     const getRowProps = (item) => {
       return {
         onMouseOver: () => this.onMouseOverRow(item),
         onMouseLeave: () => this.onMouseLeaveRow(),
       };
     };
+    let filterOptions = [];
+    if (this.props.aggregations?.event?.buckets) {
+      const buckets = this.props.aggregations.event.buckets;
+      const foundUser = buckets.findIndex((d) => d.key === ANNOTATION_EVENT_USER) > -1;
+      filterOptions = foundUser
+        ? buckets
+        : [{ key: ANNOTATION_EVENT_USER, doc_count: 0 }, ...buckets];
+    }
+    const filters = [
+      {
+        type: 'field_value_selection',
+        field: 'event',
+        name: 'Event',
+        multiSelect: 'or',
+        options: filterOptions.map((field) => ({
+          value: field.key,
+          name: field.key,
+          view: `${field.key} (${field.doc_count})`,
+        })),
+      },
+    ];
+
+    if (
+      this.props.chartDetails?.entityData?.entities &&
+      this.props.chartDetails?.entityData?.entities.length > 0
+    ) {
+      this.props.chartDetails?.entityData?.entities.forEach((entity) => {
+        if (entity.fieldType === 'partition_field') {
+          columns.push({
+            field: 'partition_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.partitionColumnName', {
+              defaultMessage: 'Partition',
+            }),
+            sortable: true,
+          });
+        }
+
+        if (entity.fieldType === 'over_field') {
+          columns.push({
+            field: 'over_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.overColumnName', {
+              defaultMessage: 'Over',
+            }),
+            sortable: true,
+          });
+        }
+        if (entity.fieldType === 'by_field') {
+          columns.push({
+            field: 'by_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.byColumnName', {
+              defaultMessage: 'By',
+            }),
+            sortable: true,
+          });
+        }
+      });
+
+      filters.push({
+        type: 'is',
+        field: CURRENT_SERIES,
+        name: i18n.translate('xpack.ml.annotationsTable.seriesOnlyFilterName', {
+          defaultMessage: 'Filter to series',
```

Comment:
When the 'Filter to series' is 'off' in the Single Metric Viewer we need to show:
1. Annotations for the series being viewed (exact match of detector index and by/over/partition fields)
2. Annotations for the job (no detector index or by/over/partition fields).

Currently it is showing also showing annotations for other series. For example, here I am viewing airline:ACA but the chart and table show annotations for airline:AAL.

![image](https://user-images.githubusercontent.com/7405507/86461255-9e8f8e00-bd21-11ea-9c22-77a54de983dc.png)

Reply:
If I hit the 'Filter to series' button, I now see this:

![image](https://user-images.githubusercontent.com/7405507/87330675-d7eda800-c530-11ea-8787-20ce5d5f8fe5.png)

Reply:
Tested after https://github.com/elastic/kibana/pull/70198/commits/487178366e43d8ebe48bed9e1acb13a7efb8da25 and now working as expected.